pyCGM_Single/tests/test_Pipelines.py

```diff
@@ -0,0 +1,517 @@
+    with pytest.raises(Exception):
+        Pipelines.transform_from_mov(data,key,clust,last_time,i)
+
+"""
+The following functions test Pipelines.segmentFinder(key,data,targetDict,segmentDict,j,missings),
+where data is an array of dictionaries of marker data,
+key is the name of the missing marker to find the segment for,
+targetDict is a dictionary of marker to segment,
+segmentDict is a dictionary of segments to marker names,
+j is the frame number the marker is missing for, 
+and missings is a dictionary indicating other missing markers.
+
+We test to ensure that missing markers are not used to reconstruct
+other missing markers. 
+
+We test the exception that is raised when the marker name does not
+exist.
+"""
+data,_ = load_data_from_file(3) #Indicates that we use files from SampleData/Sample_2/ for testing
+targetDict = target_dict()
+segmentDict = segment_dict()
+@pytest.mark.parametrize("key,data,targetDict,segmentDict,j,missings,expectedResult", [
+    ('LFHD',data,targetDict,segmentDict,10,{},
+        ['RFHD', 'RBHD', 'LBHD']),
+    ('LFHD',data,targetDict,segmentDict,10,{'LFHD':[]},
+        ['RFHD', 'RBHD', 'LBHD']),
+    ('LFHD',data,targetDict,segmentDict,10,{'RFHD':[10]},
+        ['RBHD', 'LBHD']),
+    ('LFHD',data,targetDict,segmentDict,10,{'LBHD':[10], 'RFHD':[10]},
+        ['RBHD']),
+    ('LFHD',data,targetDict,segmentDict,10,{'LBHD':[10], 'RFHD':[10], 'RBHD':[10]},
+        []),
+    ('C7',data,targetDict,segmentDict,10,{},
+        ['STRN', 'CLAV', 'T10', 'RBAK', 'RSHO', 'LSHO']),
+    ('RPSI',data,targetDict,segmentDict,10,{},
+        ['LPSI', 'LASI', 'RASI']),
+    ('LKNE',data,targetDict,segmentDict,10,{},
+        ['LTHI'])
+])
+def test_segmentFinderAccuracy(key,data,targetDict,segmentDict,j,missings,expectedResult):
+    result = Pipelines.segmentFinder(key,data,targetDict,segmentDict,j,missings)
+    assert result == expectedResult
+
+@pytest.mark.parametrize("key,data,targetDict,segmentDict,j,missings", [
+    #Test that the marker name must exist
+    ('InvalidKey',data,targetDict,segmentDict,10,{})
+])
+def test_segmentFinderExceptions(key,data,targetDict,segmentDict,j,missings):
+    with pytest.raises(Exception):
+        Pipelines.segmentFinder(key,data,targetDict,segmentDict,j,missings)
+
+"""
+The following functions test Pipelines.rigid_fill(Data, static),
+where Data is an array of dictionaries of marker data,
+and static is an array of dictionaries of static trial data.
+
+This function fills gaps for frames with missing data.
+
+We test simulate missing data by clearing six different frames
+and testing the gap filling result.
+
+We test the exception raised when the marker SACR does not
+exist in Data.
+"""
+data,static = load_data_from_file(3, calcSacrum=True) #True indicates we calculate SACR before testing
+framesToClear = [1, 10, 12, 15, 100, -1]
+for frameNum in framesToClear:
+    data['LFHD'][frameNum] = np.array([np.nan, np.nan, np.nan])
+
+result = Pipelines.rigid_fill(data, static)
```

Comment:
this should be in a function.

Reply:
Changed rigid_fill to be called inside of the test_rigid_fill_accuracy function.